x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -2939,4 +2941,69 @@ public void testNamedFunctionArgumentWithUnsupportedNamedParameterTypes() {
             );
         }
     }
+
+    public void testValidJoinPattern() {
```

Comment:
Using the index pattern makes sense for parser (vs grammar) validation however I would have expected for patterns to be rejected from the start instead of waiting to resolve them first.
That is `LOOKUP JOIN foo` works during parsing (but might fail at runtime if it's an alias pointing to multiple indices) however `LOOKUP JOIN foo*` fails at parsing since it's pattern (regardless of whether it gets resolved to 1 or more indices).